cloudinit/config/cc_set_passwords.py

```diff
@@ -45,9 +45,11 @@ def get_users_by_type(users_list: list, pw_type: str) -> list:
     )
 
 
-def _restart_ssh_daemon(distro, service):
+def _restart_ssh_daemon(distro: Distro, service: str):
     try:
-        distro.manage_service("restart", service)
+        distro.manage_service(
+            "restart", service, "--job-mode='ignore-dependencies'"
```

Comment:
Please remove the single quote around `ignore-dependencies` . Otherwise one gets:

```
> 2024-12-17 03:29:28,613 - cc_set_passwords.py[WARNING]: 'ssh_pwauth' configuration may not be applied. Cloud-init was unable to restart SSH daemon due to error: 'Unexpected error while running command.
> Command: ['systemctl', 'restart', 'sshd', "--job-mode='ignore-dependencies'"]
> Exit code: 1
> Reason: -
> Stdout:
> Stderr: Failed to restart sshd.service: Job mode 'ignore-dependencies' invalid
```